crates/starknet-devnet-core/src/starknet/get_class_impls.rs

```diff
@@ -25,12 +25,23 @@ pub fn get_class_hash_at_impl(
 }
 
 pub fn get_class_impl(
-    starknet: &mut Starknet,
+    starknet: &mut Starknet, // TODO can be regular reference?
     block_id: &BlockId,
     class_hash: ClassHash,
 ) -> DevnetResult<ContractClass> {
-    let state = starknet.get_mut_state_at(block_id)?;
-    match state.get_rpc_contract_class(&class_hash) {
+    starknet.get_block(block_id)?; // returns error if block at the id doesn't exist
+
+    // if block id is hash, convert to number; the underlying logic only works with that or tag
+    let block_tag_or_number = match block_id {
```

Comment:
It _is_ related to the linked comment:
- We work only with block numbers because of speed.
- We can only determine the block number from block hash if we have access to `Starknet.blocks` object - not accessible inside of `Starknet.rpc_contract_classes`.